docs/terms/profiler.md

```diff
@@ -1,15 +1,163 @@
+
+#### BasicDatasetProfiler
+
+The `BasicDatasetProfiler` examines a Batch of data and creates a report that answers the basic questions most data practitioners would ask about a dataset during exploratory data analysis. The Profiler reports how unique the values in the column are, as well as the percentage of empty values in it. Based on the column's type it provides a description of the column by computing a number of statistics, such as min, max, mean and median, for numeric columns, and distribution of values, when appropriate.
+
+#### BasicSuiteBuilderProfiler
+
+The `BasicSuiteBuilderProfiler` is an extension of the `BasicDatasetProfiler` that creates a set of coarse Expectations for the profiled dataset.  Configuration of this Profiler is optional; if no configuration is provided, this Profiler will create Expectations for every column in the dataset.  
+
+The values in Expectations created by the `BasicSuiteBuilderProfiler` are extremely coarse and do not correspond to the actual values in the profiled data.  Rather, the purpose of this Profiler is to generate a broad range of candidate Expectations for the profiled data, which you can then prune down to those Expectations you find relevant and edit with values based on domain knowledge.  You will definitely want to edit the values used in Expectations created by this Profiler! 
+
+#### ColumnsExistProfiler
+
+The `ColumnsExistProfiler` is an extension of the `BasicDatasetProfiler` that additionally adds an Expectation that each column found in the profiled dataset exists in the source data system. 
+
+#### JsonSchemaProfiler
+
+The `JsonSchemaProfiler` creates Expectation Suites from JSONSchema artifacts. Basic suites can be created from these specifications.
+
+:::note
+- There is not yet a notion of nested data types in Great Expectations so suites generated by a `JsonSchemaProfiler` use column map expectations.
+- A `JsonSchemaProfiler` does not traverse nested schemas and requires a top level object of type `object`.
+:::
+
+For an example of how to use the `JsonSchemaProfiler`, see our guide on [how to create a new Expectation Suite by profiling from a JsonSchema file](../guides/expectations/advanced/how_to_create_a_new_expectation_suite_by_profiling_from_a_jsonschema_file.md).
+
+### Extendable to account for domain knowledge
+
+You can also extend Profilers to capture organizational knowledge about your data. For example, a team might have a convention that all columns **named** "id" are primary keys, whereas all columns ending with the **suffix** "_id" are foreign keys. In that case, when the team using Great Expectations first encounters a new dataset that followed the convention, a Profiler could use that knowledge to add an `expect_column_values_to_be_unique` Expectation to the "id" column (but not, for example an "address_id" column).
+
+
+## API basics
+
+### How to access
```

Comment:
I would expect a lot of this section to be in separate How To Guide pages. Some of the content feels a bit repetitive with existing How Tos. I'd love to better understand what content goes in which section.

Reply:
This section was originally part of the section on Rule Based Profilers for the Profilers core concept document (and comprised a very large chunk of that document).  I felt it was important to include mostly for the discussion about the components of a rule-based profiler, which is largely overlooked in other documents (since they are focused on showing how to repeat a process more than explaining how the Rule Based Profiler works).  It also seemed applicable since Rule-Based Profilers are heavily dependent on being configured properly, so it seemed important to include the additional explanation and context for those users who came to this page because they were having difficulty following along with an existing guide for using a Rule-Based Profiler, or because they were interested in learning more about how Rule-Based Profilers work.